src/storage-client/src/controller/collection_mgmt.rs

```diff
@@ -247,34 +76,46 @@ impl CollectionManager {
     /// - Accept appends for. However, note that when appending, the
     ///   `CollectionManager` expects to be the only writer.
     pub(super) fn register_collection(&self, id: GlobalId) {
-        self.collections
-            .lock()
-            .expect("collection_mgmt panicked")
-            .insert(id);
+        let mut guard = self.collections.lock().expect("collection_mgmt panicked");
+
+        // Check if this collection is already registered.
+        if let Some((_writer, task)) = guard.get(&id) {
+            // The collection is already registered and the task is still running so nothing to do.
+            if !task.is_finished() {
+                return;
+            }
+        }
+
+        // Spawns a new task so we can write to this collection.
+        let writer_and_handle = write_task(id, self.write_handle.clone(), self.now.clone());
+        let prev = guard.insert(id, writer_and_handle);
+
+        // Double check the previous task was actually finished.
```

Comment:
What is the use case for supporting registering the same collection multiple times?

Reply:
There isn't one, I just wanted to guard against programming bugs if we ever accidentally register twice

Reply:
Added a `tracing::error!` call here, so we should be able to find out pretty quick if any scenario in production double-registers. Also added a `TODO` to change this to a panic if we never log this error in prod, how does that sound?